### DISCUSSION TITLE
Is lit support emotionjs ?

### DISCUSSION THREAD

**Discussion by happy-little-one** [lit/lit#3250] [Q&A]
```
const app = () => html`<div class=${css({ color: 'red'  })}></div>`
```

In this case, Which way will lit treate the class propetity? dynamic or static ? 

I  prefer css-in-js for my css solution, I wonder will this cause performance problems, thanks!

**Comment by augustjk** [comment]
This should work for for standalone `lit-html` as since the `css` function from `@emotion/css` returns a string to be set as a class. On update/re-render, if emotion's `css` returns the same string (which it should if the style object passed to it hasn't changed), `lit-html` wouldn't have to update anything.

I can't say with confidence what the performance impact might be. Using an expression instead of a hardcoded string as class introduces an extra check, but I'd imagine the impact would be minimal if the class is the same. This is, of course, not considering whatever performance cost of actually calling emotion's `css` would be and I have not looked into how that's implemented.

One caveat is `@emotion/css` injects style to the document head so it won't apply for elements within the shadow DOM.

Here's it working in the playground. https://lit.dev/playground/#gist=ca3cccb33d1f17dd6571a6ef7012f3d1

**Reply by happy-little-one** [reply]
seemed emotion always return a string, I guess it's ok.
